src/main/java/org/asciidoctor/asciidoclet/AsciidoctorRenderer.java

```diff
@@ -127,7 +127,28 @@ private void renderTag(Tag tag, StringBuilder buffer) {
         //print out directly
         buffer.append(tag.name());
         buffer.append(" ");
-        buffer.append(render(tag.text(), true));
+        if ("@param".equals(tag.name())) {
+            // Special handling for @param <T> tags
+            // See http://docs.oracle.com/javase/1.5.0/docs/tooldocs/windows/javadoc.html#@param
+            String parameterName = getParameterName(tag.text());
+            buffer.append(parameterName).append(' ');
+            buffer.append(render(tag.text().replace(parameterName, ""), true));
+        }
+        else {
+            buffer.append(render(tag.text(), true));
+        }
+    }
+
+    protected String getParameterName(String text) {
+        int openBracket = text.indexOf('<');
+        int closeBracket = text.indexOf('>');
+        if (openBracket != -1 && closeBracket != -1) {
+            String parameterName = text.substring(openBracket, closeBracket + 1);
+            if (parameterName.matches("\\<[a-zA-Z_$][a-zA-Z\\d_$]*\\>")) {
+                return parameterName;
+            }
```

Comment:
I think this does too much unnecessary logic. I would simply check if the text starts with '<'. If so, use the following regex to extract the type param.

```
"^<[a-zA-Z_$][a-zA-Z\\d_$]*>"
```

If it starts with `<`, there's a 99% chance this is a type param, so the optimization to look for the closing bracket isn't worth it.